freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/conversion/FmsUserConverter.java

```diff
@@ -36,11 +36,11 @@ public FmsUser toFmsUser(
     }
 
     private FmsUser createFmsUser(String workloadUsername, String firstName, String lastName) {
-        checkArgument(!Strings.isNullOrEmpty(workloadUsername));
+        checkArgument(StringUtils.isNotBlank(workloadUsername));
         FmsUser fmsUser = new FmsUser();
-        fmsUser.withName(workloadUsername);
-        fmsUser.withFirstName(StringUtils.defaultIfBlank(firstName, NONE_STRING));
-        fmsUser.withLastName(StringUtils.defaultIfBlank(lastName, NONE_STRING));
+        fmsUser.withName(StringUtils.strip(workloadUsername));
```

Comment:
This isn't needed since the workload username is generated internally and is guaranteed to contain only alphanumeric, '-' and '_'. I would prefer that the workload user name is not modified by user sync.

see https://github.infra.cloudera.com/thunderhead/thunderhead/blob/master/docs/cdp/workload_usernames.md for details on how workload usernames are generated.

Reply:
in that case it won't make any modification, so I see no harm done here by stripping it from spaces. But if you think it can cause any issue later I will revert that part

Reply:
Please revert this part.
Semantically, we want the workload username in the control plane to match exactly with the username in the workloads. It is better to pass the field through directly without any transformations, even if the transformations are not expected to make any modifications.

Reply:
sure, done. please check again